subsys/bluetooth/host/gatt.c

```diff
@@ -275,6 +276,59 @@ static void sc_process(struct k_work *work)
 	atomic_set_bit(sc->flags, SC_INDICATE_PENDING);
 }
 
+#if defined(CONFIG_BT_SETTINGS_CCC_STORE_ON_WRITE)
+static struct gatt_ccc_store {
+	struct bt_conn *conn_list[CONFIG_BT_MAX_CONN];
+	struct k_delayed_work work;
+} gatt_ccc_store;
+
+static bool gatt_ccc_is_conn_queued(struct bt_conn *conn)
+{
+	if (conn == gatt_ccc_store.conn_list[bt_conn_get_id(conn)]) {
+		return true;
+	}
+
+	return false;
+}
+
+static void gatt_ccc_conn_unqueue(struct bt_conn *conn)
+{
+	u8_t index = bt_conn_get_id(conn);
+
+	bt_conn_unref(gatt_ccc_store.conn_list[index]);
+	gatt_ccc_store.conn_list[index] = NULL;
+}
+
+static bool gatt_ccc_conn_queue_empty(void)
+{
+	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+		if (gatt_ccc_store.conn_list[i]) {
+			return false;
+		}
+	}
+
+	return true;
+}
+
+static void ccc_delayed_store(struct k_work *work)
+{
+	struct gatt_ccc_store *ccc_store =
+		CONTAINER_OF(work, struct gatt_ccc_store, work);
+
+	for (size_t i = 0; i < CONFIG_BT_MAX_CONN; i++) {
+		if (ccc_store->conn_list[i] == NULL &&
```

Comment:
Shouldn't this be != NULL? Doesn't seem like you tested this part of the code (since it'll trigger NULL-pointer dereferences).

Reply:
You are right, fixed.